File: virtusbk/src/main/java/com/dua/virtusbk/entity/Topic.java
package com.dua.virtusbk.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.hibernate.annotations.Cascade;
import org.hibernate.annotations.Type;
import org.springframework.context.annotation.Lazy;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.time.Instant;
import java.time.LocalDateTime;

@Table(name = "topics")
@Data
@NoArgsConstructor
@Entity
@ToString

public class Topic {
    public Topic(Long id) {
        this.id = id;
    }

    @Id
    @SequenceGenerator(name="topics_id_topic_seq", sequenceName="topics_id_topic_seq", allocationSize=1)
    @GeneratedValue(generator="topics_id_topic_seq")
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_topic", nullable = false)
    private Long id;

    @Column(name = "name_topic", nullable = false, length = 50)
    private String nameTopic;

    @Column(name = "description_topic", nullable = false, length = 100)
    private String descriptionTopic;

    @Column(name = "keywords_topic")
    @Lob
    @Type(type = "org.hibernate.type.TextType")
    private String keywordsTopic;

    @Column(name = "pathimg_topic", length = 200)
    private String pathimgTopic;

    @Column(name = "datereg_topic", nullable = false)
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss", iso = DateTimeFormat.ISO.DATE_TIME)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dateregTopic;

    @Column(name = "dateupdate_topic", nullable = false)
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss", iso = DateTimeFormat.ISO.DATE_TIME)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dateupdateTopic;

    @Column(name = "state_topic", nullable = false, length = 1)
    private String stateTopic;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
//    @Cascade(value={org.hibernate.annotations.CascadeType.ALL})
    @JoinColumn(name = "syllabus_id_syllabu", nullable = false)
//    @JsonIgnore// No traer toda la relación en una consulta
    private Syllabu syllabuIdSyllabu;

}

File: virtus/src/app/models/Topic.ts
import {Modules} from "./Modules";

export class Topic {
  private id: number;
  private nameTopic: string;
  private descriptionTopic: string;
  private keywordsTopic: string;
  private pathimgTopic: string;
  private dateregTopic: string;
  private dateupdateTopic: string;
  private stateTopic: string;
  private syllabuIdSyllabu: Modules;

  constructor(id: number, nameTopic: string, descriptionTopic: string, keywordsTopic: string, pathimgTopic: string, dateregTopic: string, dateupdateTopic: string, stateTopic: string) {
    this.id = id;
    this.nameTopic = nameTopic;
    this.descriptionTopic = descriptionTopic;
    this.keywordsTopic = keywordsTopic;
    this.pathimgTopic = pathimgTopic;
    this.dateregTopic = dateregTopic;
    this.dateupdateTopic = dateupdateTopic;
    this.stateTopic = stateTopic;
  }

  get _id(): number {
    return this.id;
  }

  set_id(value: number) {
    this.id = value;
  }

  get _nameTopic(): string {
    return this.nameTopic;
  }

  set_nameTopic(value: string) {
    this.nameTopic = value;
  }

  get _descriptionTopic(): string {
    return this.descriptionTopic;
  }

  set_descriptionTopic(value: string) {
    this.descriptionTopic = value;
  }

  get _keywordsTopic(): string {
    return this.keywordsTopic;
  }

  set_keywordsTopic(value: string) {
    this.keywordsTopic = value;
  }

  get _pathimgTopic(): string {
    return this.pathimgTopic;
  }

  set_pathimgTopic(value: string) {
    this.pathimgTopic = value;
  }

  get _dateregTopic(): string {
    return this.dateregTopic;
  }

  set_dateregTopic(value: string) {
    this.dateregTopic = value;
  }

  get _dateupdateTopic(): string {
    return this.dateupdateTopic;
  }

  set_dateupdateTopic(value: string) {
    this.dateupdateTopic = value;
  }

  get _stateTopic(): string {
    return this.stateTopic;
  }

  set_stateTopic(value: string) {
    this.stateTopic = value;
  }

  get _syllabuIdSyllabu(): Modules {
    return this.syllabuIdSyllabu;
  }

  set _syllabuIdSyllabu(value: Modules) {
    this.syllabuIdSyllabu = value;
  }
}
